week2/index.js

```diff
@@ -0,0 +1,213 @@
+const fs = require('fs');
+const readline = require('readline');
+let todos = [];//to be filled todo list
+
+//check if there is a parameter
+if(process.argv[2] !== undefined){
+    start();
+} else {
+    console.log('Please specify an option to use, here is the manual\n');
+    manualReader('./help.txt');
+}
+
+
+function start(){
+    switch(process.argv[2].trim().toLowerCase()){
+        case 'help':
+            manualReader('./help.txt');
+            break;
+        case 'add':
+            checkFileContent().then((item) => addTodoItem(item)).catch((err) => console.log(err));
+            break;
+        case 'list':
+            console.log('The items on your todo list: ');
+            todoFileReader('./todos.txt');
+            break;
+        case 'remove':
+            itemRemover(process.argv[3]);
+            break;
+        case 'reset':
+            itemsResetter();
+            break;
+        case 'cross':
+            crossItem(process.argv[3]);
+            break;
+        case 'uncross':
+            uncrossItem(process.argv[3]);
+            break;
+        case 'update':
+            update(process.argv[3]);
+            break;
+        default:
+            console.log('No such option! Take a look at the manual\n');
+            manualReader('./help.txt');
+            break;
+    }
+} 
+
+
+
+//  display the manual
+function manualReader(fileName){
+    fs.readFile(fileName, 'UTF-8', (err, data) => {
+        console.log(data);
+    });
+}
+
+//  display the todo file once called
+function todoFileReader(fileName){
+    fs.exists(fileName, (exists) => {
```

Comment:
Good job here - this is a different approach as compared to attempting to read the file and handle the error if the file doesn't exist.

What you've done - checking this first - is usually called 'Look Before You Leap'. https://docs.python.org/3/glossary.html#term-lbyl
The other way - just try and see what happens - is called 'Easier to ask for Forgiveness than Permission'. https://docs.python.org/3/glossary.html#term-eafp

Reply:
Hahaha I had no idea, the names are so great! 😆